src/urllib3/util/timeout.py

```diff
@@ -181,7 +192,7 @@ def clone(self):
         # detect the user default.
         return Timeout(connect=self._connect, read=self._read, total=self.total)
 
-    def start_connect(self):
+    def start_connect(self) -> Optional[float]:
```

Comment:
This function always returns `float` and is only `Optional[float]` due to `self._start_connect`'s type, should we do `self._start_connect = ret = time.monotonic(); return ret`?